quinn-proto/src/connection/mod.rs

```diff
@@ -2306,7 +2309,7 @@ where
                     self.streams.received_max_streams(dir, count)?;
                 }
                 Frame::ResetStream(frame) => {
-                    if self.streams.received_reset(frame)? {
+                    if self.streams.received_reset(frame)? && self.streams.want_send_max_data() {
```

Comment:
I think either this check should be embedded in `received_reset`, or (perhaps preferably for consistency) `received_reset`'s `bool` return should be removed as redundant.

Reply:
Yes, that can definitely be merged. As far as I understand the main intention of the boolean return value at the moment is to signal whether there is something to send - so moving it inside would be ok. The main counterargument is that all other checks at the moment are also outside on the connection level. But with the discussed refactoring that point would go away.